Cargo.toml

```diff
@@ -49,6 +52,51 @@ cairo-lang-semantic = { version = "~2.5.4" }
 cairo-lang-sierra = { version = "~2.5.4" }
 cairo-lang-syntax = { version = "~2.5.4" }
 
+[dependencies]
+anyhow = { workspace = true }
+assert_matches = { workspace = true }
+async-stream = { workspace = true }
+base64 = { workspace = true }
+bitvec = { workspace = true }
+blockifier = { workspace = true }
+cairo-type-derive = { workspace = true }
+cairo-vm = { workspace = true }
+futures = { workspace = true }
+futures-util = { workspace = true }
+getset = { workspace = true }
+heck = { workspace = true }
+hex = { workspace = true }
+indexmap = { workspace = true }
+indoc = { workspace = true }
+lazy_static = { workspace = true }
+log = { workspace = true }
+num-bigint = { workspace = true }
+num-integer = { workspace = true }
+num-traits = { workspace = true }
+regex = { workspace = true }
+reqwest = { workspace = true }
+serde = { workspace = true }
+serde_json = { workspace = true }
+serde_with = { workspace = true }
+serde_yaml = { workspace = true }
+starknet-crypto = { workspace = true }
+starknet_api = { workspace = true }
+thiserror = { workspace = true }
+tokio = { workspace = true }
+tokio-stream = { workspace = true }
+uuid = { workspace = true }
+zip = { workspace = true }
+
+# These dependencies pin cairo-lang requirement versions because cairo-vm does not do it
+cairo-lang-starknet = { workspace = true }
+cairo-lang-casm = { workspace = true }
+cairo-lang-defs = { workspace = true }
+cairo-lang-diagnostics = { workspace = true }
+cairo-lang-filesystem = { workspace = true }
+cairo-lang-semantic = { workspace = true }
+cairo-lang-sierra = { workspace = true }
+cairo-lang-syntax = { workspace = true }
```

Comment:
I hadn't thought about this duplication. This limits the interest of the workspace. I'm not sure if this is the right way to go. What do you think?

Reply:
Agreed, it's a nuisance, and isn't very valuable with only a couple minor sub-crates. I'm rather indifferent.

These dependencies would be duplicated in any case, but it would be cleaner if the crate `snos` was in a subdir. We could make that change while we don't have many PRs open... :shrug: